core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -881,6 +884,16 @@ public void withStartupAttempts(int attempts) {
         this.startupAttempts = attempts;
     }
 
+    @Override
+    public void close() {
+        stop();
+    }
+
+    public SELF withCustomizer(Consumer<CreateContainerCmd> hook) {
```

Comment:
I liked pre/post, but actually doesn't it make it confusing wrt. _what_ it is before or after?

This change is post- testcontainer's configuration step, but pre- creation of the container. So actually it seems to be confusing if we're not careful 😂 

Some other ideas for the name: `withConfigurationCustomizer`, `withConfigModifier` or maybe even just `with(Consumer<CreateContainerCmd> hook)`..? (The latter would need heavier documentation so as to be more discoverable though)

I'm not sure though 🤷‍♂️